From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Sat, 8 May 2021 23:30:22 +0200
Subject: [PATCH] tcpops: use fixup_igp_all() instead of fixup_igp_null()

- needed for functions with many parameters

(cherry picked from commit 02240711239149e2f5c4890a70ab158d10fa8187)
---
 src/modules/tcpops/tcpops_mod.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/modules/tcpops/tcpops_mod.c b/src/modules/tcpops/tcpops_mod.c
index b2d2fbe..23036fd 100644
--- a/src/modules/tcpops/tcpops_mod.c
+++ b/src/modules/tcpops/tcpops_mod.c
@@ -78,27 +78,27 @@ static pv_export_t mod_pvs[] = {
 
 static cmd_export_t cmds[]={
 	{"tcp_keepalive_enable", (cmd_function)w_tcp_keepalive_enable4, 4,
-			fixup_igp_null, fixup_free_igp_null, ANY_ROUTE},
+			fixup_igp_all, fixup_free_igp_all, ANY_ROUTE},
 	{"tcp_keepalive_enable", (cmd_function)w_tcp_keepalive_enable3, 3,
-			fixup_igp_null, fixup_free_igp_null, REQUEST_ROUTE|ONREPLY_ROUTE},
+			fixup_igp_all, fixup_free_igp_all, REQUEST_ROUTE|ONREPLY_ROUTE},
 	{"tcp_keepalive_disable", (cmd_function)w_tcp_keepalive_disable1, 1,
-			fixup_igp_null, fixup_free_igp_null, ANY_ROUTE},
+			fixup_igp_all, fixup_free_igp_all, ANY_ROUTE},
 	{"tcp_keepalive_disable", (cmd_function)w_tcp_keepalive_disable0, 0,
 			0, 0, REQUEST_ROUTE|ONREPLY_ROUTE},
 	{"tcp_set_connection_lifetime", (cmd_function)w_tcpops_set_connection_lifetime2, 2,
-			fixup_igp_null, fixup_free_igp_null, ANY_ROUTE},
+			fixup_igp_all, fixup_free_igp_all, ANY_ROUTE},
 	{"tcp_set_connection_lifetime", (cmd_function)w_tcpops_set_connection_lifetime1, 1,
-			fixup_igp_null, fixup_free_igp_null, REQUEST_ROUTE|ONREPLY_ROUTE},
+			fixup_igp_all, fixup_free_igp_all, REQUEST_ROUTE|ONREPLY_ROUTE},
 	{"tcp_enable_closed_event", (cmd_function)w_tcpops_enable_closed_event1, 1,
-			fixup_igp_null, fixup_free_igp_null, ANY_ROUTE},
+			fixup_igp_all, fixup_free_igp_all, ANY_ROUTE},
 	{"tcp_enable_closed_event", (cmd_function)w_tcpops_enable_closed_event0, 0,
 			0, 0, REQUEST_ROUTE|ONREPLY_ROUTE},
 	{"tcp_conid_state", (cmd_function)w_tcp_conid_state, 1,
-			fixup_igp_null, fixup_free_igp_null, ANY_ROUTE},
+			fixup_igp_all, fixup_free_igp_all, ANY_ROUTE},
 	{"tcp_get_conid", (cmd_function)w_tcp_get_conid, 2,
 			fixup_spve_pvar, fixup_free_spve_pvar, ANY_ROUTE},
 	{"tcp_conid_alive", (cmd_function)w_tcp_conid_alive, 1,
-			fixup_igp_null, fixup_free_igp_null, ANY_ROUTE},
+			fixup_igp_all, fixup_free_igp_all, ANY_ROUTE},
 	{"tcp_set_otcpid", (cmd_function)w_tcp_set_otcpid, 1,
 			fixup_igp_all, fixup_free_igp_all, ANY_ROUTE},
 	{"tcp_set_otcpid_flag", (cmd_function)w_tcp_set_otcpid_flag, 1,
